### PR TITLE
New Event: Atmos Leak

### PR DESCRIPTION
**Pull request by Leshana**
In this event some sort of gas leak occours! A random area on the station is chosen to experience a leak of a randomly chosen gas!
* By default, areas with any people in them are protected, so it won't just kill someone.
* A warning announcement is given shortly before, so there is some warning.
* Dorms, Shuttles, Holodeck and Supermatter are protected.
* At higher severity, adds worse gasses and possibility of high/low temperature.

### CODE REVIEW

**Comment by SkowronX** [comment]
Damn it, shipyard.

File: code/modules/events/atmos_leak.dm [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+	var/gas_choices = list("carbon_dioxide", "sleeping_agent") // Annoying
+	if(severity >= EVENT_LEVEL_MODERATE)
+		gas_choices += "phoron" // Dangerous
+	if(severity >= EVENT_LEVEL_MAJOR)
+		gas_choices += "volatile_fuel" // Dangerous and no default atmos setup!
+	gas_type = pick(gas_choices)
+
+	// Assemble areas that all exists (See DM reference if you are confused about loop labels)
+	var/list/area/grand_list_of_areas = list()
+	looping_station_areas:
```
Comment (Iamgoofball) [inline code/modules/events/atmos_leak.dm]: aha what the fuck no

File: code/modules/events/atmos_leak.dm [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+			// Otherwise add it and all subtypes that exist on the map to our grand list
+			for(var/areapath in typesof(parentpath))
+				var/area/A = locate(areapath) // Check if it actually exists
+				if(istype(A) && A.z in using_map.player_levels)
+					grand_list_of_areas += A
+
+	// Okay, now lets try and pick a target! Lets try 10 times, otherwise give up
+	for(var/i in 1 to 10)
+		var/area/A = pick(grand_list_of_areas)
+		if(is_area_occupied(A))
```
Comment (Iamgoofball) [inline code/modules/events/atmos_leak.dm]: this is dumb as shit

Comment (Iamgoofball) [inline code/modules/events/atmos_leak.dm]: who cares if someone's in there

**Comment by Anewbe** [comment]
@Iamgoofball If you're going to comment on our stuff, don't be a dick.

File: code/modules/events/atmos_leak.dm [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+		/area/engineering/engine_room
+	)
+
+// Decide which area will be targeted!
+/datum/event/atmos_leak/setup()
+	var/gas_choices = list("carbon_dioxide", "sleeping_agent") // Annoying
+	if(severity >= EVENT_LEVEL_MODERATE)
+		gas_choices += "phoron" // Dangerous
+	if(severity >= EVENT_LEVEL_MAJOR)
+		gas_choices += "volatile_fuel" // Dangerous and no default atmos setup!
```
Comment (Neerti) [inline code/modules/events/atmos_leak.dm]: Will this gas function properly, and how is it different from phoron?

Comment (Leshana) [inline code/modules/events/atmos_leak.dm]: Yup, I didn't make it up, its a real gas defined in XGM like the other gasses.   It is actually the gas for when welder fuel or stuff is spilled and burned.
It burns like phoron, the difference is that it doesn't show up as named on atmos scan, it shows up as "other" so its more likely techs will assume its okay to use welders and initiate fun(tm), That's why its only an option on the highest severity level.
Please see attached gif for what volatile_fuel does.
ezgif com-resize

**Pull request merged**